presto-cli/src/main/java/com/facebook/presto/cli/StatusPrinter.java

```diff
@@ -154,20 +156,22 @@ private void updateScreen(WarningsPrinter warningsPrinter)
     private String autoFormatMetricValue(RuntimeUnit unit, long value)
     {
         if (unit == RuntimeUnit.NANO) {
-            return formatTime(Duration.succinctNanos(value));
+            return formatTime(succinctNanos(value));
         }
         if (unit == RuntimeUnit.BYTE) {
             return formatDataSize(bytes(value), true);
         }
         return formatCount(value); // NONE
     }
 
-    public void printFinalInfo()
+    public void printFinalInfo(long clientStopTimestamp)
     {
-        Duration wallTime = nanosSince(start);
-
         QueryStatusInfo results = client.finalStatusInfo();
         StatementStats stats = results.getStats();
+        Duration serverSideWallTime = succinctDuration(stats.getElapsedTimeMillis(), MILLISECONDS);
+        //We can't use the nanosSince(start) for client side latency since that will include time the user spent viewing the results in the pager
+        //Instead we use the caller provided clientStopTimestamp for accurate client side latency
+        Duration clientSideWallTime = succinctNanos(clientStopTimestamp - start);
```

Comment:
Bug : if query fails with an error, clientStopTimestamp will not have a value set (will be 0), so the succintNanos calls will fail